core/src/main/java/hudson/model/Queue.java

```diff
@@ -539,7 +539,7 @@ public synchronized WaitingItem schedule(Task p, int quietPeriod, List<Action> a
                 shouldScheduleItem |= action.shouldSchedule(actions);
     		}
     		for (QueueAction action: Util.filter(actions,QueueAction.class)) {
-                shouldScheduleItem |= action.shouldSchedule(item.getActions());
+                shouldScheduleItem |= action.shouldSchedule((new ArrayList<Action>(item.getAllActions())));
```

Comment:
not sure why transient actions should be involved here?  @jglick

Reply:
Why not?